server/controllers/conversation.js

```diff
@@ -0,0 +1,67 @@
+const Conversation = require("../models/Conversation");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /conversations
+// @desc Create new conversation
+// @access Public
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { dogOwnerId, dogSitterId } = req.body;
+
+  try {
+    const conversation = await Conversation.create({
+      dogOwnerId,
+      dogSitterId
+    });
+    if (conversation)
+      res.status(201).json({ success: "Successfully created conversation" });
+    else
+      res.status(400).json({ error: "Could not make a conversation" });
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+// @route GET /conversations
+// @desc Get all conversations
+// @access Public
+exports.getAllConversations = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const allConversations = await Conversation.find({
+    $or: [{ "petSitter": userId }, { "petOwner": userId }]
+  }).populate("recentMessage");
+
+  if (allConversations) {
```

Comment:
```
    res.status(200).json({
      success: {
        conversations: allConversations
      }
    });
```
We don't need an if/else clause here as the `find()` query returns an empty list when no matching records are found (an empty list is a truthy value). Ideally we want to forward this empty list to the FE to have a consistent response format :)